app/policies/aal3_policy.rb

```diff
@@ -1,50 +1,72 @@
 class AAL3Policy
   AAL3_METHODS = %w[webauthn piv_cac].freeze
 
-  attr_reader :mfa_policy, :auth_method, :service_provider, :aal_level_requested
+  attr_reader :mfa_policy, :auth_method, :service_provider, :aal_level_requested,
+              :piv_cac_requested
 
   def initialize(
     user:,
     service_provider:,
     auth_method:,
-    aal_level_requested:
+    aal_level_requested:,
+    piv_cac_requested:
   )
+    @user = user
     @mfa_policy = MfaPolicy.new(user)
     @auth_method = auth_method
     @service_provider = service_provider
     @aal_level_requested = aal_level_requested
+    @piv_cac_requested = piv_cac_requested
   end
 
   def aal3_required?
+    return false if piv_cac_required?
     aal3_requested? || aal3_configured?
   end
 
   def aal3_used?
+    return false if piv_cac_required?
```

Comment:
ditto for this.... I'd expect someting like restricting `AAL3_METHODS` to just `piv_cac` if `piv_cac_required?`

Reply:
Maybe we should update the calling site here? Or change the method? This is causing some cognitive dissonance for me, since `pivcac` is a kind of aal3.... in HSPD12 mode, we need some other method name?